selenium.common.exceptions.JavascriptException: Message: javascript error: Cannot read properties of null (reading 'shadowRoot')

**Describe the bug**
Chrome and Edge error when `wait_for_downloads()` is called from headless mode.

**_The screenshot returns a blank image._**  :expressionless:
![img](https://user-images.githubusercontent.com/51935281/152193842-092c8252-dab0-482e-9e8f-98e0c64816ab.png)

[Same problem - StackOverflow](https://stackoverflow.com/questions/62444031/selenium-unable-to-access-chrome-internal-pages-in-headless-mode)

```python
import os
import pytest

from botcity.web import WebBot, Browser

project_path = os.path.abspath('')


@pytest.fixture
def web() -> WebBot:
    web = WebBot()
    web.headless = True
    web.browser = Browser.CHROME
    web.driver_path = os.path.join(project_path, 'web-drivers', 'chromedriver.exe')
    yield web

    web.stop_browser()

def test_wait_for_downloads(web: WebBot):
    web.browse('https://sabnzbd.org/tests/internetspeed/50MB.bin')
    web.wait(1000)

    web.wait_for_downloads()
    # assert
```

```python
D:\BotCity\projetos\dev\web-test\venv\Scripts\python.exe C:\Users\luizf\AppData\Roaming\JetBrains\IntelliJIdea2021.3\plugins\python\helpers\pycharm\_jb_pytest_runner.py --target test_browser.py::test_wait_for_downloads
Testing started at 12:26 ...
Launching pytest with arguments test_browser.py::test_wait_for_downloads --no-header --no-summary -q in D:\BotCity\projetos\dev\web-test

============================= test session starts =============================
collecting ... collected 1 item

test_browser.py::test_wait_for_downloads FAILED                          [100%]
test_browser.py:258 (test_wait_for_downloads)
web = <botcity.web.bot.WebBot object at 0x000002802D10FA30>

    def test_wait_for_downloads(web: WebBot):
        # web.browse(os.path.join(project_path, 'web', 'index.html'))
        web.browse('https://sabnzbd.org/tests/internetspeed/50MB.bin')
        web.wait(1000)
    
>       web.wait_for_downloads()

test_browser.py:267: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\dev\botcity-framework-web-python\botcity\web\bot.py:1013: in wait_for_downloads
    WebDriverWait(self._driver, timeout/1000, 1).until(wait_method)
venv\lib\site-packages\selenium\webdriver\support\wait.py:71: in until
    value = method(self._driver)
..\..\dev\botcity-framework-web-python\botcity\web\browsers\chrome.py:115: in wait_for_downloads
    return driver.execute_script("""
venv\lib\site-packages\selenium\webdriver\remote\webdriver.py:634: in execute_script
    return self.execute(command, {
venv\lib\site-packages\selenium\webdriver\remote\webdriver.py:321: in execute
    self.error_handler.check_response(response)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x000002802D10EE00>
response = {'status': 500, 'value': '{"value":{"error":"javascript error","message":"javascript error: Cannot read properties of ...unk [0x76806739+25]\\n\\tRtlGetFullPathName_UEx [0x771D8AFF+1215]\\n\\tRtlGetFullPathName_UEx [0x771D8ACD+1165]\\n"}}'}

    def check_response(self, response):
        """
        Checks that a JSON response from the WebDriver does not have an error.
    
        :Args:
         - response - The JSON response from the WebDriver server as a dictionary
           object.
    
        :Raises: If the response contains an error message.
        """
        status = response.get('status', None)
        if status is None or status == ErrorCode.SUCCESS:
            return
        value = None
        message = response.get("message", "")
        screen = response.get("screen", "")
        stacktrace = None
        if isinstance(status, int):
            value_json = response.get('value', None)
            if value_json and isinstance(value_json, basestring):
                import json
                try:
                    value = json.loads(value_json)
                    if len(value.keys()) == 1:
                        value = value['value']
                    status = value.get('error', None)
                    if status is None:
                        status = value["status"]
                        message = value["value"]
                        if not isinstance(message, basestring):
                            value = message
                            message = message.get('message')
                    else:
                        message = value.get('message', None)
                except ValueError:
                    pass
    
        exception_class = ErrorInResponseException
        if status in ErrorCode.NO_SUCH_ELEMENT:
            exception_class = NoSuchElementException
        elif status in ErrorCode.NO_SUCH_FRAME:
            exception_class = NoSuchFrameException
        elif status in ErrorCode.NO_SUCH_WINDOW:
            exception_class = NoSuchWindowException
        elif status in ErrorCode.STALE_ELEMENT_REFERENCE:
            exception_class = StaleElementReferenceException
        elif status in ErrorCode.ELEMENT_NOT_VISIBLE:
            exception_class = ElementNotVisibleException
        elif status in ErrorCode.INVALID_ELEMENT_STATE:
            exception_class = InvalidElementStateException
        elif status in ErrorCode.INVALID_SELECTOR \
                or status in ErrorCode.INVALID_XPATH_SELECTOR \
                or status in ErrorCode.INVALID_XPATH_SELECTOR_RETURN_TYPER:
            exception_class = InvalidSelectorException
        elif status in ErrorCode.ELEMENT_IS_NOT_SELECTABLE:
            exception_class = ElementNotSelectableException
        elif status in ErrorCode.ELEMENT_NOT_INTERACTABLE:
            exception_class = ElementNotInteractableException
        elif status in ErrorCode.INVALID_COOKIE_DOMAIN:
            exception_class = InvalidCookieDomainException
        elif status in ErrorCode.UNABLE_TO_SET_COOKIE:
            exception_class = UnableToSetCookieException
        elif status in ErrorCode.TIMEOUT:
            exception_class = TimeoutException
        elif status in ErrorCode.SCRIPT_TIMEOUT:
            exception_class = TimeoutException
        elif status in ErrorCode.UNKNOWN_ERROR:
            exception_class = WebDriverException
        elif status in ErrorCode.UNEXPECTED_ALERT_OPEN:
            exception_class = UnexpectedAlertPresentException
        elif status in ErrorCode.NO_ALERT_OPEN:
            exception_class = NoAlertPresentException
        elif status in ErrorCode.IME_NOT_AVAILABLE:
            exception_class = ImeNotAvailableException
        elif status in ErrorCode.IME_ENGINE_ACTIVATION_FAILED:
            exception_class = ImeActivationFailedException
        elif status in ErrorCode.MOVE_TARGET_OUT_OF_BOUNDS:
            exception_class = MoveTargetOutOfBoundsException
        elif status in ErrorCode.JAVASCRIPT_ERROR:
            exception_class = JavascriptException
        elif status in ErrorCode.SESSION_NOT_CREATED:
            exception_class = SessionNotCreatedException
        elif status in ErrorCode.INVALID_ARGUMENT:
            exception_class = InvalidArgumentException
        elif status in ErrorCode.NO_SUCH_COOKIE:
            exception_class = NoSuchCookieException
        elif status in ErrorCode.UNABLE_TO_CAPTURE_SCREEN:
            exception_class = ScreenshotException
        elif status in ErrorCode.ELEMENT_CLICK_INTERCEPTED:
            exception_class = ElementClickInterceptedException
        elif status in ErrorCode.INSECURE_CERTIFICATE:
            exception_class = InsecureCertificateException
        elif status in ErrorCode.INVALID_COORDINATES:
            exception_class = InvalidCoordinatesException
        elif status in ErrorCode.INVALID_SESSION_ID:
            exception_class = InvalidSessionIdException
        elif status in ErrorCode.UNKNOWN_METHOD:
            exception_class = UnknownMethodException
        else:
            exception_class = WebDriverException
        if value == '' or value is None:
            value = response['value']
        if isinstance(value, basestring):
            if exception_class == ErrorInResponseException:
                raise exception_class(response, value)
            raise exception_class(value)
        if message == "" and 'message' in value:
            message = value['message']
    
        screen = None
        if 'screen' in value:
            screen = value['screen']
    
        stacktrace = None
        if 'stackTrace' in value and value['stackTrace']:
            stacktrace = []
            try:
                for frame in value['stackTrace']:
                    line = self._value_or_default(frame, 'lineNumber', '')
                    file = self._value_or_default(frame, 'fileName', '<anonymous>')
                    if line:
                        file = "%s:%s" % (file, line)
                    meth = self._value_or_default(frame, 'methodName', '<anonymous>')
                    if 'className' in frame:
                        meth = "%s.%s" % (frame['className'], meth)
                    msg = "    at %s (%s)"
                    msg = msg % (meth, file)
                    stacktrace.append(msg)
            except TypeError:
                pass
        if exception_class == ErrorInResponseException:
            raise exception_class(response, message)
        elif exception_class == UnexpectedAlertPresentException:
            alert_text = None
            if 'data' in value:
                alert_text = value['data'].get('text')
            elif 'alert' in value:
                alert_text = value['alert'].get('text')
            raise exception_class(message, screen, stacktrace, alert_text)
>       raise exception_class(message, screen, stacktrace)
E       selenium.common.exceptions.JavascriptException: Message: javascript error: Cannot read properties of null (reading 'shadowRoot')
E         (Session info: headless chrome=97.0.4692.99)

venv\lib\site-packages\selenium\webdriver\remote\errorhandler.py:242: JavascriptException


============================== 1 failed in 5.47s ==============================

Process finished with exit code 1
```

**Expected behavior**
Wait to finish download

**Steps to Reproduce**
```python
def test_wait_for_downloads():
    project_path = os.path.abspath('')
    web = WebBot()
    web.headless = True
    web.browser = Browser.CHROME
    web.driver_path = os.path.join(project_path, 'web-drivers', 'chromedriver.exe')
    
    web.browse('https://sabnzbd.org/tests/internetspeed/50MB.bin')
    web.wait(1000)

    web.wait_for_downloads()
    # assert
```

**My Platform**
    * OS: Windows 11
    * Python Version: 3.10.2
    * Packages Version: `botcity-framework-web  0.3.1+7.g1b7d79f`

--- a/botcity/web/bot.py
+++ b/botcity/web/bot.py
@@ -1022,6 +1022,17 @@ class WebBot(BaseBot):
         Args:
             timeout (int, optional): Timeout in millis. Defaults to 120000.
         """
+        if self.browser in [Browser.CHROME, Browser.EDGE] and self.headless:
+            start_time = time.time()
+            while True:
+                elapsed_time = (time.time() - start_time) * 1000
+                if elapsed_time > timeout:
+                    return False
+                downloads_count = self.get_file_count(self.download_folder_path, ".crdownload")
+                if downloads_count == 0:
+                    return True
+                self.sleep(config.DEFAULT_SLEEP_AFTER_ACTION)
+
         wait_method = BROWSER_CONFIGS.get(self.browser).get("wait_for_downloads")
         # waits for all the files to be completed
         WebDriverWait(self._driver, timeout/1000, 1).until(wait_method)

